source/extensions/quic_listeners/quiche/codec_impl.cc

```diff
@@ -1,41 +1,93 @@
 #include "extensions/quic_listeners/quiche/codec_impl.h"
 
+#include "extensions/quic_listeners/quiche/envoy_quic_client_stream.h"
 #include "extensions/quic_listeners/quiche/envoy_quic_server_stream.h"
 
 namespace Envoy {
 namespace Quic {
 
 bool QuicHttpConnectionImplBase::wantsToWrite() { return quic_session_.HasDataToWrite(); }
 
+void QuicHttpConnectionImplBase::runWatermarkCallbacksForEachStream(
+    quic::QuicSmallMap<quic::QuicStreamId, std::unique_ptr<quic::QuicStream>, 10>& stream_map,
+    bool high_watermark) {
+  for (auto& it : stream_map) {
+    if (!it.second->is_static()) {
+      auto stream = dynamic_cast<EnvoyQuicStream*>(it.second.get());
```

Comment:
nit: can you have a method called something like QuicheStreamToEnvoyStream() which does the dynamic cast and has a comment about why we have the dynamic cast and then use that everywhere we are currently doing the dynamic cast? I think I understand why it's done but it would be good to have it be self-documenting?

Reply:
done